src/main/java/net/ripe/rpki/rsyncit/rrdp/RrdpFetcher.java

```diff
@@ -103,16 +109,16 @@ public FetchResult fetchObjectsEx() {
             }
 
             long begin = System.currentTimeMillis();
-            final byte[] snapshotContent = loadSnapshot(snapshotUrl, desiredSnapshotHash);
+            var snapshot = loadSnapshot(snapshotUrl, desiredSnapshotHash);
```

Comment:
What about this scenario (note that this is a concurrency/race condition, so "they are kind of close so this is unlikely" is not enough):
  * `node-i` retrieves `serial x` at `t0`
  * CDN cache expires, serves `new notification.xml` that refers to a snapshot that is a minute newer.
  * `node-i+1` retrieves `serial x+1` at `t0 + sigma`

The files that differ between the snapshots will be out of sync forever.

Reply:
(Think this would work if we used deltas btw, because then files are in the same file)

Reply:
Yeah, that makes sense.